source/common/http/conn_manager_impl.cc

```diff
@@ -679,13 +679,15 @@ void ConnectionManagerImpl::maybeDrainDueToPrematureResets() {
       runtime_.snapshot().getInteger(ConnectionManagerImpl::PrematureResetTotalStreamCountKey, 500);
 
   if (closed_non_internally_destroyed_requests_ < limit) {
-    return;
-  }
-
-  if (static_cast<double>(number_premature_stream_resets_) /
-          closed_non_internally_destroyed_requests_ <
-      .5) {
-    return;
+    if (number_premature_stream_resets_ < .5 * limit) {
```

Comment:
This will be faster as integer math
```suggestion
    if (number_premature_stream_resets_ < limit / 2) {
```

Reply:
Done (thought I did '* 2' on the lhs instead of '/ 2' on the RHS in case limit is odd).